boa/src/object/internal_methods.rs

```diff
@@ -881,6 +881,67 @@ impl GcObject {
         Ok(list)
     }
 
+    /// It is used to iterate over names of object's keys.
+    ///
+    /// More information:
+    /// - [EcmaScript reference][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-enumerableownpropertynames
+    pub(crate) fn enumerable_own_property_names(
+        &self,
+        kind: PropertyNameKind,
+        context: &mut Context,
+    ) -> Result<Vec<JsValue>> {
+        // 1. Assert: Type(O) is Object.
+        // 2. Let ownKeys be ? O.[[OwnPropertyKeys]]().
+        let own_keys = self.own_property_keys();
+        // 3. Let properties be a new empty List.
+        let mut properties = vec![];
+
+        // 4. For each element key of ownKeys, do
+        for key in own_keys {
+            // a. If Type(key) is String, then
+            if let PropertyKey::String(key_str) = &key {
+                // i. Let desc be ? O.[[GetOwnProperty]](key).
+                let desc = self.__get_own_property__(&key);
+                // ii. If desc is not undefined and desc.[[Enumerable]] is true, then
+                if let Some(desc) = desc {
+                    if desc.expect_enumerable() {
+                        // 1. If kind is key, append key to properties.
+                        if let PropertyNameKind::Key = kind {
+                            properties.push(JsValue::String(key_str.clone()))
+                        }
+                        // 2. Else,
+                        else {
+                            // a. Let value be ? Get(O, key).
+                            let value = self.get(key.clone(), context)?;
+                            // b. If kind is value, append value to properties.
+                            if let PropertyNameKind::Value = kind {
+                                properties.push(value)
+                            }
+                            // c. Else,
+                            else {
+                                // i. Assert: kind is key+value.
+                                // ii. Let entry be ! CreateArrayFromList(« key, value »).
+                                // TODO: Implement https://tc39.es/ecma262/#sec-createarrayfromlist
+                                let entry = JsValue::new_object(context);
+                                let key_val = JsValue::String(key_str.clone());
+                                entry.add(&key_val, context)?;
+                                entry.add(&value, context)?;
+
+                                // iii. Append entry to properties.
+                                properties.push(entry);
```

Comment:
```suggestion
                                let entry = Array::array_create(2, None, context)?;
                                entry.create_data_property_or_throw(0, key_str.clone(), context)?;
                                entry.create_data_property_or_throw(1, value, context)?;
                                // iii. Append entry to properties.
                                properties.push(entry.into());
```
Workaround until we implement `CreateArrayFromList`.